go_bazel_test not handling data entries now

This is because of 6683d0614ed4170efe596a4d110178ca3a91b243. With it reverted, `go_bazel_test` works correctly.

If you add an entry to the `data` arg of a `go_bazel_test` (say `//go/tools/gopackagesdriver:gopackagesdriver`) , your test will fail with:

```
INFO: From Testing //tests/core/gopackages:gopackages_test:
==================== Test output for //tests/core/gopackages:gopackages_test:
error: unexpected file: go/tools/gopackagesdriver/darwin_amd64_stripped/gopackagesdriver
panic: runtime error: invalid memory address or nil pointer dereference
================================================================================
```

--- a/go/tools/bazel_testing/bazel_testing.go
+++ b/go/tools/bazel_testing/bazel_testing.go
@@ -106,7 +106,7 @@ func TestMain(m *testing.M, args Args) {
 	}
 	if beginFiles >= 0 {
 		files = os.Args[beginFiles+1 : endFiles-1]
-		os.Args = append(os.Args[:beginFiles], os.Args[endFiles+1:]...)
+		os.Args = append(os.Args[:beginFiles:beginFiles], os.Args[endFiles+1:]...)
 	}
 
 	flag.Parse()

